openslides/utils/cache_providers.py

```diff
@@ -204,6 +209,23 @@ def get_change_id_cache_key(self) -> str:
         async with get_connection() as redis:
             return await redis.hgetall(cache_key)
 
+    async def get_collection_data(
+        self, collection: str, user_id: Optional[int] = None
+    ) -> Dict[bytes, bytes]:
+        """
+        Returns all elements for a collection from the cache.
+        """
+        if user_id is None:
+            cache_key = self.get_full_data_cache_key()
+        else:
+            cache_key = self.get_restricted_data_cache_key(user_id)
```

Comment:
Why do you add this second branch/case für restricted data? This pull request does not use it.

Reply:
Thats true. I added if for the future. And this is a bad thing to do.

Reply:
Do you want to add the fix for the restricted data cache here directly?

Reply:
No. If you don't like the different cache keys at this place, I can remove it. But I would prefer to merge it as it is.